### DISCUSSION TITLE
Multilayer networks

### DISCUSSION THREAD

**Discussion by jbruun** [mapequation/infomap#307] [Q&A] [answered]
Some time ago, it was possible to input separate pajek files each as individual layers in a multilayer network:
./Infomap network1.net network2.net [... networkN.net] output/

This seems to not be possible anymore? This will mean that I will need to specify a multilayer network locally, which is of course doable. 

I guess that making a multilayer network without interlayer connections would be the same, or am I missing something? If so, what?

Also, I will join the (small) quire of people who lament the loss of --overlapping. I was/am aware of the assumptions of that particular method, but the results the times I used it were actually meaningful. So perhaps the model could be reintroduced? As well as the awesome multi-network-input option :)

**Comment by mrosvall** [comment, marked as answer]
Correct, you want to use the multilayer format with only intralayer links. Example from https://www.mapequation.org/infomap/#InputMultilayer

```
# A multilayer network using *Intra format
# Figure 5
*Vertices 5
# node_id name
1 "i" 
2 "j" 
3 "k" 
4 "l" 
5 "m" 
*Intra
# layer_id node_id node_id weight
1 1 4 1 
1 4 1 1 
1 1 5 1 
1 5 1 1 
1 4 5 1 
1 5 4 1 
2 1 2 1 
2 2 1 1 
2 1 3 1 
2 3 1 1 
2 2 3 1 
2 3 2 1 
```

We are working on something better and more principled than the --overlapping option. Meanwhile, would you be interested in a script that reads a standard network and outputs a second-order network (with first-order information) in state format https://www.mapequation.org/infomap/#InputStates ?

**Reply by antoneri** [reply]
As a compromise for your first question, I've added the file [merge-multilayer.py](https://github.com/mapequation/infomap/blob/master/utils/merge-multilayer.py).

Run it with `python utils/merge-multilayer.py network1 network2 networkN outfile`

Hope this helps!

**Reply by jbruun** [reply]
Thanks to both for your answers and good work!

**Reply by jbruun** [reply]
Hah, I just read the last part of your reply, Martin. I can see that Anton already posted a solution to the multiple networks idea. If possible, it would be nice with a script that reads a standard networkand outputs a second-order network.

**Reply by antoneri** [reply]
Here you go! [network-to-states.py](https://github.com/mapequation/infomap/blob/master/utils/network-to-states.py)

Usage:
`network-to-states.py [-d] network_file states_file`
where -d should be used if your input network is directed.

You should also let Infomap know if the input network is directed when you run it.

Note that this creates quite large files for large-ish networks. On a test network with around 7 000 nodes and 630 000 links, the resulting state network has 630 000 states and 157 million links!

Let me know how it works!
